contracts/ValidatorSet/modules/Staking/Staking.sol

```diff
@@ -138,10 +130,11 @@ abstract contract Staking is IStaking, ValidatorSetBase, BalanceState, Withdrawa
     ) internal returns (uint256 validatorStakeLeft, uint256 totalStakeLeft) {
         uint256 totalStake = balanceOf(validator);
         uint256 delegation = rewardPool.totalDelegationOf(validator);
-        if (amount > totalStake || amount + delegation > totalStake)
-            revert StakeRequirement({src: "unstake", msg: "INSUFFICIENT_BALANCE"});
-
         uint256 validatorStake = totalStake - delegation;
+        // if (amount > totalStake || amount + delegation > totalStake)
+        if (amount > validatorStake) revert StakeRequirement({src: "unstake", msg: "INSUFFICIENT_BALANCE"});
+
+        // uint256 validatorStake = totalStake - delegation;
         totalStakeLeft = totalStake - amount;
```

Comment:
totalStakeLeft is unused

Reply:
Yeah, but it is part of the named return params.